iwyu_ast_util.cc

```diff
@@ -942,12 +950,20 @@ bool IsFriendDecl(const Decl* decl) {
   return decl->getFriendObjectKind() != Decl::FOK_None;
 }
 
+bool IsExplicitInstantiation(const clang::Decl* decl) {
+  clang::TemplateSpecializationKind kind = GetTemplateSpecializationKind(decl);
+  return kind == clang::TSK_ExplicitInstantiationDeclaration ||
+         kind == clang::TSK_ExplicitInstantiationDefinition;
+}
+
 bool IsForwardDecl(const NamedDecl* decl) {
-  if (const auto* record_decl = dyn_cast<RecordDecl>(decl)) {
+  if (const CXXRecordDecl* record_decl = DynCastFrom(decl)) {
```

Comment:
Hmm, `DynCastFrom` predates `auto` and I think it's clearer to use LLVM's `dyn_cast` and friends with `auto` than leaning on this hack.

To be perfectly honest, I think it's slightly easier to read the `DynCastFrom` line, but I had to spend a considerable amount of time understand how it worked the first time I saw it. `auto` is well-understood.